lldb/test/API/linux/aarch64/sme_only_registers/TestSMEOnlyRegisters.py

```diff
@@ -0,0 +1,553 @@
+"""
+Check handling of registers on an AArch64 Linux system that only has SME. As
+opposed to SVE and SME. Check register access and restoration after expression
+evaluation.
+"""
+
+from enum import Enum
+from pprint import pprint
+from functools import lru_cache
+from itertools import permutations
+import lldb
+from lldbsuite.test.decorators import *
+from lldbsuite.test.lldbtest import *
+from lldbsuite.test import lldbutil
+from itertools import cycle
+
+
+class Mode(Enum):
+    SIMD = 0
+    SSVE = 2
+
+    def __str__(self):
+        return "streaming" if self == Mode.SSVE else "simd"
+
+
+class ZA(Enum):
+    ON = 1
+    OFF = 2
+
+    def __str__(self):
+        return "on" if self == ZA.ON else "off"
+
+
+class ByteVector(object):
+    def __init__(self, data):
+        self.data = data
+
+    def __repr__(self):
+        return "{" + " ".join([f"0x{b:02x}" for b in self.data]) + "}"
+
+    def as_bytes(self):
+        return self.data
+
+
+class HexValue(object):
+    # Assume all values are 64-bit, but some display as 32-bit, like fpcr.
+    def __init__(self, value, repr_size=8):
+        self.value = value
+        # In bytes
+        self.repr_size = repr_size
+
+    def __repr__(self):
+        return f"0x{self.value:0{self.repr_size*2}x}"
+
+    def as_bytes(self):
+        data = []
+        v = self.value
+        # Little endian order.
+        for i in range(8):
+            data.append(v & 0xFF)
+            v >>= 8
+
+        return data
+
+
+class SVESIMDRegistersTestCase(TestBase):
+    def reg_names(self, prefix, count):
+        return [f"{prefix}{n}" for n in range(count)]
+
+    def expected_registers(self, svl_b, mode, za):
+        register_values = []
+
+        if mode == Mode.SIMD:
+            # In streaming mode we have real V registers and no Z registers.
```

Comment:
non-streaming mode?  On a system with only FEAT_SME, not FEAT_SVE, like the Apple M4/M5.